File: app/src/main/java/com/abishov/hexocat/home/trending/TrendingPagerFragment.java
package com.abishov.hexocat.home.trending;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.abishov.hexocat.R;
import com.abishov.hexocat.commons.views.BaseFragment;

public final class TrendingPagerFragment extends BaseFragment {

    public static TrendingPagerFragment create() {
        return new TrendingPagerFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
            @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.trending_pager_view, container, false);
    }
}

File: hexocat-android/app/src/main/java/com/abishov/hexocat/home/trending/TrendingPagerFragment.kt
package com.abishov.hexocat.home.trending

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.abishov.hexocat.R
import com.abishov.hexocat.common.views.BaseFragment

fun create(): TrendingPagerFragment {
  return TrendingPagerFragment()
}

class TrendingPagerFragment : BaseFragment() {

  override fun onCreateView(
    inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
  ) = inflater.inflate(R.layout.trending_pager_view, container, false)

  // @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    val trendingPagerView = view as TrendingPagerView

    trendingPagerView.setAdapter(
      TrendingPagerAdapter(context!!, childFragmentManager)
    )
  }
}
